packages/principled-forms/src/field/index.ts

```diff
@@ -20,6 +20,14 @@ const _validate = <T>(field: Field<T>): Validated[] => {
   return rule(...field.validators)(field.value);
 };
 
+type OnInvalid = (reason: string) => (Unvalidated | Invalid);
+
+const reduceValidities = (validities: Validated[], onInvalid: OnInvalid = Validity.Invalid.because) => {
```

Comment:
We need a better name, but extracting this is good!

Whenever I write a reducer, I ask "What am I reducing *from* and *to*, and try to write the name accordingly. So e.g. if it's something I'll call inline, I'll name it so it can be something like `validities.reduce(toSingleValidity)`.

Reply:
Here, maybe `validities.reduce(toAnyInvalid)` or something?